internal/app/machined/pkg/controllers/network/link_config.go

```diff
@@ -276,58 +282,78 @@ func (ctrl *LinkConfigController) parseMachineConfiguration(logger *zap.Logger,
 		}
 
 		for _, linkName := range device.Bond().Interfaces() {
-			if _, exists := bondedLinks[linkName]; exists {
+			if bondName, exists := bondedLinks[linkName]; exists && bondName != device.Interface() {
 				logger.Sugar().Warnf("link %q is included into more than two bonds", linkName)
 			}
 
 			bondedLinks[linkName] = device.Interface()
 		}
 	}
 
+	linkMap := map[string]*network.LinkSpecSpec{}
+
 	for _, device := range cfgProvider.Machine().Network().Devices() {
 		if device.Ignore() {
 			continue
 		}
 
-		link := network.LinkSpecSpec{
-			Name:        device.Interface(),
-			MTU:         uint32(device.MTU()),
-			Up:          true,
-			ConfigLayer: network.ConfigMachineConfiguration,
+		if _, exists := linkMap[device.Interface()]; !exists {
+			linkMap[device.Interface()] = &network.LinkSpecSpec{
+				Name:        device.Interface(),
+				Up:          true,
+				ConfigLayer: network.ConfigMachineConfiguration,
+			}
 		}
 
-		if bondName := bondedLinks[device.Interface()]; bondName != "" {
-			bondSlave(&link, bondName)
+		if device.MTU() != 0 {
+			linkMap[device.Interface()].MTU = uint32(device.MTU())
 		}
 
 		if device.Bond() != nil {
-			if err := bondMaster(&link, device.Bond()); err != nil {
+			if err := bondMaster(linkMap[device.Interface()], device.Bond()); err != nil {
 				logger.Error("error parsing bond config", zap.Error(err))
 			}
 		}
 
 		if device.WireguardConfig() != nil {
-			if err := wireguardLink(&link, device.WireguardConfig()); err != nil {
+			if err := wireguardLink(linkMap[device.Interface()], device.WireguardConfig()); err != nil {
 				logger.Error("error parsing wireguard config", zap.Error(err))
 			}
 		}
 
 		if device.Dummy() {
-			dummyLink(&link)
+			dummyLink(linkMap[device.Interface()])
 		}
 
 		for _, vlan := range device.Vlans() {
-			links = append(links, vlanLink(device.Interface(), vlan))
+			vlanSpec := vlanLink(device.Interface(), vlan)
+
+			linkMap[vlanSpec.Name] = &vlanSpec
 		}
+	}
+
+	for slaveName, bondName := range bondedLinks {
```

Comment:
enslave bond interfaces even if they're not mentioned at the top level in the machine config